### ISSUE TITLE
Images colorspace conversion to sRGB

### ISSUE THREAD

**Issue by cogat** [ic-labs/django-icekit#210]
`glamkit-imagetools` has a function which converts a given image to the sRGB color space, meaning that images with a different color space can be converted and don't appear too different when the profile is stripped for e.g. thumbnails and IIIF operations.

The question is, when is the right time to do the sRGB conversion? My instinctive answer is: upon thumbnail creation, rather than on upload.

So, let's do that, unless there's a compelling reason to convert the original uploaded image.

**Comment by jmurty** [comment]
I agree that we should leave sRGB conversion as late as possible. Aside from the extra complexity of doing this work on upload, unless I misunderstand it is a data-losing operation that I don't think we should apply to original images.

When reviewing the the _easy-thumbnails_ library for inspiration for the IIIF Image API I learned that thumbnails generated for colour images (not black-and-white) are automatically converted by this library into the RGB colourspace. I applied the same approach for the IIIF Image API, see https://github.com/ic-labs/django-icekit/blob/develop/icekit/plugins/iiif/views.py#L202

So in short, the RGB conversion is already being done when needed while leaving the original images intact. I don't think we need to change this approach.

Unless, that is, the _glamkit-imagetools_ does more sophisticated colour space conversion that the basic approach linked above?

**Comment by cogat** [comment]
@jmurty yep I think there's a subtlety here - I used 'colour space' to mean 'colour profile'. Converting (from CMYK) to RGB colourspace isn't the same as converting the embedded colour profile to sRGB, a specific standard defined for consumer monitors. Last I checked (in SFMOMA), EasyThumbnails doesn't do any colour profile conversion, it just strips it in the output to reduce filesize.

**Comment by jmurty** [comment]
NOTE: The conversion performed by *icekit-imagetools* is more sophisticated than the basic RGB colourspace conversion we are already doing. See https://github.com/ixc/icekit-imagetools/blob/develop/imagetools/utils/srgb.py

**Comment by cogat** [comment]
On a quick look through the list, this is the only relevant easy-thumbs ticket (ie converting to sRGB would solve this)
https://github.com/SmileyChris/easy-thumbnails/issues/366

**Comment by jmurty** [comment]
Looking at the above, to do full sRGB colour profile conversion (or just retention) we will need to:
* add sRGB handling to the IIIF Image API implementation, based on the code in *icekit-imagetools* (though probably adapted to be less reliant of images always being in local files)
* investigate whether _easy-thumbnails_ does already, or can be made to, do the sRGB conversion/retention for us. If not, we may need to monkey-patch it. The main problem seems to be including the ICC profile when writing the image, e.g. `img.save(outfile, icc_profile=SRGB_BYTES`

**Comment by cogat** [comment]
_easy-thumbnails_ doesn't do color profile conversion by default, unless it's started doing it since we built SFMOMA.

I propose that we add a `THUMBNAIL_PROCESSORS` processor that converts images to sRGB using the `glamkit-imagetools` process, perhaps in the presence of an `srgb` option.

(There's also this simpler approach from DrMeers: https://stackoverflow.com/questions/11041044/convert-jpg-from-adobergb-to-srgb-using-pil, but it may only work with Adobe RGB source profiles, not arbitrary profiles).

**Comment by cogat** [comment]
![srgb profile stripped](https://user-images.githubusercontent.com/55459/28067703-bba5fe70-6685-11e7-8d93-1255b44531a0.jpg)
![with srgb](https://user-images.githubusercontent.com/55459/28067704-bbf5a812-6685-11e7-80dd-e54e60dc245a.jpg)
![adobe rgb profile stripped](https://user-images.githubusercontent.com/55459/28067705-bc298704-6685-11e7-8ed1-3cc30bbe26f6.jpg)
![with adobe rgb](https://user-images.githubusercontent.com/55459/28067706-bc5c9fae-6685-11e7-993c-5705757efe91.jpg)

Here are 4 examples, from https://www.sfmoma.org/artwork/2005.102 . If the file is created with Adobe sRGB the colour looks flat if the profile is stripped (as easy-thumbnails will do) or even if it isn't stripped and viewed on mobile (as mobile devices ignore colour profile). Converting the file to sRGB results in minimal difference on consumer screens if the profile is stripped or ignored.

**Comment by cogat** [comment]
I have made an easythumbnails image processor based on the util above that converts an image to sRGB colour profile and RGB colour space so that there is the minimum of visual disruption when the colour profile is dropped. Add it to the beginning of THUMBNAIL_PROCESSORS as per https://easy-thumbnails.readthedocs.io/en/latest/ref/processors/#custom-processors.

https://gist.github.com/cogat/fbdfeb099564ef0cdda8491b0d65557c

I think this can replace the default colorspace processor.
